client.go

```diff
@@ -76,8 +77,18 @@ func (c *realClient) tryConnect() error {
 	if c.conn != nil && c.conn.State() != grpc.Shutdown {
 		c.conn.Close()
 	}
+	if c.addrGetter != nil {
+		c.addr = c.addrGetter()
+	}
+	if c.addr == nil {
+		return fmt.Errorf("No server address is available.")
+	}
```

Comment:
Don't we want to handle the case where there initially isn't a metricd client and addrGetter returns nil?